### PR TITLE
Avoid tons of boxing.

### PR DESCRIPTION
**Pull request by EricArndt**
The ref param boxes this int. This is called lots of times which results in a tons of objects that have to be cleaned up causing stress on the GC. Code is uglier but the performance is needed.

### CODE REVIEW

File: Tibialyzer/MainFormReadMemory.cs [hunk 1]
```diff
@@ -245,12 +228,60 @@ public class ParseMemoryResults {
...
                     if (i > array.Length - 6) break;
 
-                    if (!checkRange(array, '0', '9', ref i)) continue;
-                    if (!checkRange(array, '0', '9', ref i)) continue;
-                    if (!checkDigit(array, ':', ref i)) continue;
-                    if (!checkRange(array, '0', '9', ref i)) continue;
-                    if (!checkRange(array, '0', '9', ref i)) continue;
-                    if (!(checkDigit(array, ' ', ref i) || checkDigit(array, ':', ref i))) continue;
+                    if (array[i] >= '0' && array[i] <= '9')
+                    {
```
Comment (mathiasbynens) [inline Tibialyzer/MainFormReadMemory.cs]: Please follow the existing coding style, i.e. put braces on the same line.

**Comment by EricArndt** [comment]
For this change I will definitely follow the coding style of the existing code in the file. I have some more changes I am testing for even better performance. Might make a separate PR for that though.

May I ask why you prefer this style? It goes against the recommended style guidelines specified by Microsoft and the default formatting in Visual Studio. Would you mind if I changed the formatting for the whole project in a different PR? There are other refactorings such as separating UI from behavior that I'd like to do as well. Is that something you'd be open to?

**Comment by Mytherin** [comment]
Thanks for your commit. The reason I use this style is just personal preference. I don't like putting every bracket on a new line, it wastes space and makes code less readable in my opinion. I'd prefer to keep the formatting this way. 

Refactoring the code is something I'm in the process of doing; I think it would be counterproductive if we both work on that at the same time. I should finish refactoring sometime this weekend, after that if you still don't like the structure of the code you can refactor away.

**Comment by EricArndt** [comment]
What is the focus of the refactoring? I have some additional changes I may want to merge in soon, but I'll hold off if that will cause a messy merge.

**Comment by Mytherin** [comment]
Currently I'm mainly cleaning up the MainForm class, which has way too much functionality merged into one huge class. I'm moving functionality to new classes that each hold a specific part of the functionality so there's a better split.

**Comment by EricArndt** [comment]
I'll finish my new PR today or tomorrow and let you decide if it'll merge cleanly. Sound OK?

**Comment by mathiasbynens** [comment]
@EricArndt Tip: it’s easier to work on a separate branch for each patch (i.e. never on `master`). This allows you to add follow-up commits to existing pull requests (based on code review, for example) and to easily work on multiple separate patches.

**Comment by EricArndt** [comment]
Will do. I'm not very familiar with the PR mechanism.

**Pull request merged**